dotCMS/src/main/java/com/dotmarketing/business/UserFactoryLiferayImpl.java

```diff
@@ -182,9 +182,40 @@ public User loadByUserByEmail(String email) throws DotDataException, DotSecurity
 			throw new DotDataException(e.getMessage(), e);
 		}
 	}
-	
+
+	@Override
+	public long getCountUsersByName(String filter) throws DotDataException {
+		filter = SQLUtil.sanitizeParameter(filter);
+		DotConnect dotConnect = new DotConnect();
+		boolean isFilteredByName = UtilMethods.isSet(filter);
+		filter = (isFilteredByName ? filter : "");
+		StringBuffer baseSql = new StringBuffer("select count(*) as count from user_ where companyid = ? and userid <> 'system' ");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the synchronized class "StringBuffer" by an unsynchronized one such as "StringBuilder". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)